torch/utils/_pytree.py

```diff
@@ -243,38 +262,48 @@ def _odict_unflatten(
     return OrderedDict((key, value) for key, value in zip(context, values))
 
 
-_register_pytree_node(
-    dict,
-    _dict_flatten,
-    _dict_unflatten,
-    serialized_type_name="builtins.dict",
-)
-_register_pytree_node(
-    list,
-    _list_flatten,
-    _list_unflatten,
-    serialized_type_name="builtins.list",
-)
-_register_pytree_node(
-    tuple,
-    _tuple_flatten,
-    _tuple_unflatten,
-    serialized_type_name="builtins.tuple",
-)
-_register_pytree_node(
-    namedtuple,
-    _namedtuple_flatten,
-    _namedtuple_unflatten,
-    to_dumpable_context=_namedtuple_serialize,
-    from_dumpable_context=_namedtuple_deserialize,
-    serialized_type_name="collections.namedtuple",
-)
-_register_pytree_node(
-    OrderedDict,
-    _odict_flatten,
-    _odict_unflatten,
-    serialized_type_name="collections.OrderedDict",
-)
+SUPPORTED_NODES: Dict[Type[Any], NodeDef] = {
+    dict: NodeDef(dict, _dict_flatten, _dict_unflatten),
+    list: NodeDef(list, _list_flatten, _list_unflatten),
+    tuple: NodeDef(tuple, _tuple_flatten, _tuple_unflatten),
+    namedtuple: NodeDef(namedtuple, _namedtuple_flatten, _namedtuple_unflatten),  # type: ignore[dict-item,arg-type]
+    OrderedDict: NodeDef(OrderedDict, _odict_flatten, _odict_unflatten),
+}
+SUPPORTED_SERIALIZED_TYPES: Dict[Type[Any], _SerializeNodeDef] = {
+    dict: _SerializeNodeDef(
```

Comment:
Probably should use a `python_pytree._private_register_pytree_node` helper function  (mentioned above) to avoid repetition